fe/fe-core/src/main/java/com/starrocks/meta/lock/Locker.java

```diff
@@ -138,23 +139,64 @@ public boolean isWriteLockHeldByCurrentThread(Database database) {
     /**
      * Before the new version of LockManager is fully enabled, it is used to be compatible with the original db lock logic.
      */
-    public void lockTable(Database database, Table table, LockType lockType) {
+    public void lockTables(Database database, List<Long> tableList, LockType lockType) {
         if (Config.use_lock_manager) {
-            assert table != null;
-            lock(table.getId(), lockType, 0);
+            assert !tableList.isEmpty();
+            if (lockType == LockType.WRITE) {
+                this.lock(database.getId(), LockType.INTENTION_EXCLUSIVE, 0);
+            } else if (lockType == LockType.READ) {
+                this.lock(database.getId(), LockType.INTENTION_SHARED, 0);
+            }
+
+            Collections.sort(tableList);
+            for (Long rid : tableList) {
+                this.lock(rid, lockType, 0);
+            }
         } else {
             //Fallback to db lock
             lockDatabase(database, lockType);
         }
     }
 
+    public boolean tryLockTables(Database database, List<Long> tableList, LockType lockType, long timeout) {
+        if (Config.use_lock_manager) {
+            assert !tableList.isEmpty();
+            try {
+                if (lockType == LockType.WRITE) {
+                    this.lock(database.getId(), LockType.INTENTION_EXCLUSIVE, timeout);
```

Comment:
This will lock the corresponding db implicityly, but seems we need another api: `public boolean tryLockTables(List<Long> tableList, LockType lockType, long timeout) {}`, because we may have the following logic,
```
lock_db_IX
// do some checks about db
lock_tables_X
// do some updates
unlock_db_IX
unlock_tables_X
```

Reply:
At present, this is just compatibility logic. The process you mentioned can be easily pieced together based on the lock interface.